target/linux/mediatek/image/filogic.mk

```diff
@@ -869,6 +869,17 @@ define Device/ubnt_unifi-6-plus
 endef
 TARGET_DEVICES += ubnt_unifi-6-plus
 
+define Device/wavlink_wl-wn586x3
+  DEVICE_VENDOR := Wavlink
+  DEVICE_MODEL := WL-WN586X3
+  DEVICE_DTS := mt7981b-wavlink-wl-wn586x3
+  DEVICE_DTS_DIR := ../dts
+  DEVICE_DTS_LOADADDR := 0x47000000
+  IMAGE_SIZE := 15424k
+  DEVICE_PACKAGES := kmod-mt7981-firmware
```

Comment:
I wanted to suggest adding `mt7981-wo-firmware` too. But this is spi and I'm not sure...

Reply:
Please let me know if I should add this. Thanks.

Reply:
wo-firmware can be installed by the end user, so let's keep this as is.

P.S. You can add `-uboot-envtools` to save a bit more free space.

Reply:
I think you should add `mt7981-wo-firmware`. It is needed to offload wifi and we add it for the other mt7981 devices.

Reply:
Done.